R/class-Statistic.R

```diff
@@ -0,0 +1,83 @@
+check_statistic_with_confidence <- function(object) {
+    errors <- character()
+    value <- object@value
+    pvalue <- object@pvalue
+    ciMin <- object@confidenceInterval@minimum
+    ciMax <- object@confidenceInterval@maximum
+    ciLevel <- object@confidenceLevel
+
+    if (length(value) != 1) {
+        msg <- "The slot `value` must have a single value."
+        errors <- c(errors, msg)
+    } else
+
+    if (!is.na(pvalue)) {
+      if (length(pvalue) != 1) {
+        msg <- "The slot `pvalue` must have a single value."
+        errors <- c(errors, msg)
+      } else if (pvalue < 0 || pvalue > 1) {
+        msg <- "Provided p-value is invalid."
+        errors <- c(errors, msg)
+      }
+    }  
+
+    # false for NaN, while is.na is TRUE for NaN
+    is.NA <- function(object) { ifelse(is.na(object), ifelse(is.nan(object), FALSE, TRUE), FALSE) }
```

Comment:
idk.. i considered it but tbh in 5+ years on this project and 7+ working in R i dont think ive needed this until today. if it becomes a regular problem ill move and export it, but otherwise i dont want to risk confusion between is.na and is.NA